### PR TITLE
add roaring_bitmap_convert_to_all_bitset

### PR DESCRIPTION
**Pull request by feng-y**
Add roaring_bitmap_convert_to_all_bitset, in tens of thousands of massive search to improve the search efficiency

### CODE REVIEW

File: include/roaring/roaring.h [hunk 1]
```diff
@@ -800,6 +800,13 @@ uint32_t roaring_bitmap_maximum(const roaring_bitmap_t *r);
 void roaring_bitmap_statistics(const roaring_bitmap_t *r,
                                roaring_statistics_t *stat);
 
+/**
+ * (For expert users who seek high performance.)
+ *
+ * Convert every container to bitset, for huge contians actions.
```
Comment (lemire) [inline include/roaring/roaring.h:806]:
```suggestion
 * Convert every container to bitset, for large contains actions. 
 * Note that the memory usage may increase. Furthermore,
 * the bitmap is left in an unexpected state.
 * You should not attempt to modify the resulting bitmap, or serialize
 * it. The bitmap should only be used for contains operations.
```

**Comment by Dr-Emann** [comment]
Per what @lemire suggested, this invalidates preconditions that a lot of code in CRoaring will expect to always hold. I'm pretty sure destruction doesn't depend on containers being of the expected type for their size, but we should double check that if you haven't. It might also be worthwhile to add a function to convert back, to enable other uses of the bitmap again.

**Comment by lemire** [comment]
@Dr-Emann is correct.


It is maybe worth noting that we have fast functions for when you need to do a lot of contains operations...

```C++
bool roaring_bitmap_contains_bulk(const roaring_bitmap_t *r,
                                  roaring_bulk_context_t *context,
                                  uint32_t val);
```

Another sensible alternative to this PR would be to have a function that produces a bitset: https://github.com/lemire/cbitset

A true bitset would be even more efficient than what can be achieved with this PR (a genuine bitset has very low overhead).

**Comment by lemire** [comment]
@Dr-Emann @feng-y  

Can you check out the following alternative...

https://github.com/RoaringBitmap/CRoaring/pull/448

It should provide even better performance, with much less confusion, because it uses a whole other data structure.

**Comment by lemire** [comment]
Dropping this because we have...

https://github.com/RoaringBitmap/CRoaring/pull/448


@feng-y If PR 448 does not meet your needs, please provide data, discussion, report.

**Pull request closed without merging**